core/CMakeLists.txt

```diff
@@ -49,7 +51,7 @@ add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/include/RGitCommit.h
     ${CMAKE_BINARY_DIR}/RGitCommit.h.tmp
 )
 
-add_custom_target(gitcommit ALL DEPENDS ${CMAKE_BINARY_DIR}/RGitCommit.h.tmp ${CMAKE_BINARY_DIR}/ginclude/RGitCommit.h ${CMAKE_BINARY_DIR}/include/RGitCommit.h ${CMAKE_BINARY_DIR}/ginclude/RConfigure.h ${CMAKE_BINARY_DIR}/include/RConfigure.h)
+add_custom_target(gitcommit ALL DEPENDS ${CMAKE_BINARY_DIR}/ginclude/RGitCommit.h ${CMAKE_BINARY_DIR}/ginclude/RConfigure.h)
```

Comment:
@Axel-Naumann Why you remove includes, which should be copied into `${CMAKE_BINARY_DIR}/include`:
`${CMAKE_BINARY_DIR}/include/RGitCommit.h` and `${CMAKE_BINARY_DIR}/include/RConfigure.h`?

It just guarantees that these very basic includes are there before anything else will be compiled.
I propose to remove only `${CMAKE_BINARY_DIR}/include/RConfigure.h` and specify additional dependency 
`add_dependencies(BaseTROOT gitcommit rconfigure)`

Reply:
Axel fixed here exactly the same problem which is breaking 6.22 when user is compiling root with make -j1: there is no RConfigure.h at the moment of generation G_Core target.

Reply:
Yes but Sergey wants to guarantee that `ginclude/Foo` and `include/Foo` are identical at all times. I amended the commit and re-introduced the dependency of BaseTROOT on `include/Foo` to keep that behavior.